sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/src/ParallelSecretLoader.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Security.KeyVault.Secrets;
+
+namespace Azure.Extensions.AspNetCore.Configuration.Secrets
+{
+    internal class ParallelSecretLoader: IDisposable
+    {
+        private const int ParallelismLevel = 32;
+        private readonly SecretClient _client;
+        private readonly SemaphoreSlim _semaphore;
```

Comment:
Thought: could just use a `ConcurrentQueue` and avoid the housekeeping here.

Reply:
Can you explain, please? I don't know a way to use CQ to limit parallelism.

Reply:
As noted below, you have a small pool of threads pull work off a queue. I've written/helped write several pipelines that use that approach (mostly setup engines / chainers). The number of threads in the pool is limited, so worker queues are bounded.

Reply:
I see. I tried to avoid having to manage my own thread pool, for now, especially considering that async completions would be executed on the real thread pool anyway.